### ISSUE TITLE
NoMethodError: super called outside of method on upgrade

### ISSUE THREAD

**Issue by martinos** [cldwalker/boson#10]
I have upgraded boson to 0.2.3 and I get the following error:

```
NoMethodError: super called outside of method
    from /usr/lib/ruby/gems/1.8/gems/boson-0.2.3/lib/boson/scientist.rb:76
    from /usr/lib/ruby/gems/1.8/gems/boson-0.2.3/lib/boson/scientist.rb:86:in `call'
    from /usr/lib/ruby/gems/1.8/gems/boson-0.2.3/lib/boson/scientist.rb:86:in `call_original_command'
    from /usr/lib/ruby/gems/1.8/gems/boson-0.2.3/lib/boson/scientist.rb:94:in `translate_and_render'
    from /usr/lib/ruby/gems/1.8/gems/boson-0.2.3/lib/boson/scientist.rb:76:in `libraries'
```

It does the same thing with version 0.2.2.

Do you have an idea about the source of the problem ?

**Comment by cldwalker** [comment]
I haven't experienced that error. What operating system and ruby version are you on? What boson command did you run to get that error? What boson libraries did you have when you got this error, perhaps the ones at http://github.com/martinos/martinos-boson?

**Comment by martinos** [comment]
I am running:
Linux appsrv01 2.6.30-gentoo-r4
ruby 1.8.7 (2010-01-10 patchlevel 249) [i686-linux]

I have also deleted my old .boson directory and I have no bosonfile. So it's not a problem from a library. I have also removed my RUBYOPT environment variable. 

I have removed all previous boson, hirb and alias gems before reinstalling from scratch.

I have the problem for commands even with core ones like libraries and commands.

**Comment by cldwalker** [comment]
It seems this is a 1.8.7 issue. On ruby 1.8.6 and 1.9.1, I can run boson fine without a ~/.boson. I'm on a mac but I don't think that's the issue. When you get a chance could you try your system on 1.8.6 or 1.9.1? When I get around to it, I'll look at boson on 1.8.7. If you haven't run multiple ruby versions on your system, rvm makes this fairly easy: http://rvm.beginrescueend.com/

**Comment by martinos** [comment]
It think it might not be a problem with ruby 1.8.7 since it's what I run on my Mac and I use Boson 0.2.3 without any issue.

Anyway I will try to run 1.8.6 on the production server (the computer with the issue).

**Comment by martinos** [comment]
I tried 1.8.7 with boson 0.2.3 on Ubuntu Linux and I never had any problem. So it seem it's something with gentoo distribution.

**Comment by cldwalker** [comment]
Bizarre. Perhaps it is gentoo's 1.8.7.

I've run boson on 1.8.7 and everything works fine. All tests pass when I run `testrb test/*.rb`. If you're interested in digging deeper I'd suggest running the tests. You'll need to install these gems for testing:

```
 gem install jeremymcanally-context jeremymcanally-matchy --source http://gems.github.com
 gem install mocha
```

**Comment by CodeSturgeon** [comment]
I'm having this problem on Snow Leopard with Ruby 1.8.7 from MacPorts...
    ruby 1.8.7 (2010-01-10 patchlevel 249) [i686-darwin10]

New install. No libraries loaded.

I downloaded the repo (92e7f96ea53e808eae6466a2fd872fd29c75f951) and ran the suggested tests. 277 tests, 129 assertions, 2 failures, 37 errors.

Here is the head output including the first failure and first error. The rest are all the same.

<pre>
Loaded suite [test/argument_inspector_test.rb, test/bin_runner_test.rb, test/comment_inspector_test.rb, test/file_library_test.rb, test/loader_test.rb, test/manager_test.rb, test/method_inspector_test.rb, test/option_parser_test.rb, test/options_test.rb, test/pipes_test.rb, test/repo_index_test.rb, test/repo_test.rb, test/runner_test.rb, test/scientist_test.rb, test/test_helper.rb, test/util_test.rb]
Started
..................Error: super called outside of method
FError: super called outside of method
...........Error: super called outside of method
..Error: super called outside of method
...............................................................................................................................................................................................E.EEEEEEEEE..FE..EE.EEEEEEEEEEEEEEEEEEEEEEEE..........
Finished in 0.515079 seconds.

  1) Failure:
test: at commandline global option takes value with whitespace(Boson::BinRunnerTest::TestAtCommandline2160080140)
    [./test/bin_runner_test.rb:58:in `test: at commandline global option takes value with whitespace'
     /opt/local/lib/ruby/gems/1.8/gems/jeremymcanally-context-0.5.5/lib/context/suite.rb:32:in `run'
     /opt/local/lib/ruby/gems/1.8/gems/jeremymcanally-context-0.5.5/lib/context/suite.rb:30:in `each'
     /opt/local/lib/ruby/gems/1.8/gems/jeremymcanally-context-0.5.5/lib/context/suite.rb:30:in `run'
     /opt/local/lib/ruby/gems/1.8/gems/jeremymcanally-context-0.5.5/lib/context/suite.rb:32:in `run'
     /opt/local/lib/ruby/gems/1.8/gems/jeremymcanally-context-0.5.5/lib/context/suite.rb:30:in `each'
     /opt/local/lib/ruby/gems/1.8/gems/jeremymcanally-context-0.5.5/lib/context/suite.rb:30:in `run'
     /opt/local/lib/ruby/gems/1.8/gems/jeremymcanally-context-0.5.5/lib/context/suite.rb:32:in `run'
     /opt/local/lib/ruby/gems/1.8/gems/jeremymcanally-context-0.5.5/lib/context/suite.rb:30:in `each'
     /opt/local/lib/ruby/gems/1.8/gems/jeremymcanally-context-0.5.5/lib/context/suite.rb:30:in `run']:
not all expectations were satisfied
unsatisfied expectations:
- expected exactly once, not yet invoked: Boson::View.render()
satisfied expectations:
- allowed any number of times, not yet invoked: Hirb.enable(any_parameters)

  2) Error:
test: optionless command renders(Boson::ScientistTest):
NoMethodError: super called outside of method
    ./test/../lib/boson/scientist.rb:76
    ./test/../lib/boson/scientist.rb:86:in `call'
    ./test/../lib/boson/scientist.rb:86:in `call_original_command'
    ./test/../lib/boson/scientist.rb:94:in `translate_and_render'
    ./test/../lib/boson/scientist.rb:76:in `blah'
    ./test/scientist_test.rb:35:in `send'
    ./test/scientist_test.rb:35:in `command'
    ./test/scientist_test.rb:275:in `test: optionless command renders'
    /opt/local/lib/ruby/gems/1.8/gems/mocha-0.9.8/lib/mocha/integration/test_unit/ruby_version_186_and_above.rb:19:in `__send__'
    /opt/local/lib/ruby/gems/1.8/gems/mocha-0.9.8/lib/mocha/integration/test_unit/ruby_version_186_and_above.rb:19:in `run'
    /opt/local/lib/ruby/gems/1.8/gems/jeremymcanally-context-0.5.5/lib/context/suite.rb:32:in `run'
    /opt/local/lib/ruby/gems/1.8/gems/jeremymcanally-context-0.5.5/lib/context/suite.rb:30:in `each'
    /opt/local/lib/ruby/gems/1.8/gems/jeremymcanally-context-0.5.5/lib/context/suite.rb:30:in `run'
    /opt/local/lib/ruby/gems/1.8/gems/jeremymcanally-context-0.5.5/lib/context/suite.rb:32:in `run'
    /opt/local/lib/ruby/gems/1.8/gems/jeremymcanally-context-0.5.5/lib/context/suite.rb:30:in `each'
    /opt/local/lib/ruby/gems/1.8/gems/jeremymcanally-context-0.5.5/lib/context/suite.rb:30:in `run'
    /opt/local/lib/ruby/gems/1.8/gems/jeremymcanally-context-0.5.5/lib/context/suite.rb:32:in `run'
    /opt/local/lib/ruby/gems/1.8/gems/jeremymcanally-context-0.5.5/lib/context/suite.rb:30:in `each'
    /opt/local/lib/ruby/gems/1.8/gems/jeremymcanally-context-0.5.5/lib/context/suite.rb:30:in `run'
</pre>

**Comment by cldwalker** [comment]
Thanks for the error report. On 1.8.7 patchlevel 174 installed with rvm, I'm not getting this error but some other ones instead. When I switch boson to a better test gem i.e. bacon, I'll look into this. Since I don't use 1.8.7 much I can't guarantee this will be soon.

**Comment by cldwalker** [comment]
This commit in the latest release fixed this:891cb01149b390
This bug was actually being caused by a ruby bug: http://redmine.ruby-lang.org/issues/show/2537.

**Comment by CodeSturgeon** [comment]
Yay! I just swung back to some projects I was thinking boson could be perfect for. 

Just to confirm:
    277 tests, 548 assertions, 0 failures, 0 errors
